extensions/amp-fit-text/0.2/amp-fit-text.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FitText} from './fit-text';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+
+import {dict} from '../../../src/utils/object';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-fit-text';
+
+class AmpFitText extends PreactBaseElement {
+  /** @override */
+  init() {
+    const attributeOb = new MutationObserver(() => {
```

Comment:
Is this needed in AMP mode? The only mutation should be via `mutatedAttributesCallback`, which will already cause a rerender.

Reply:
Would about Bento mode?

Reply:
I think we should instead do this:

1. Move `new MutationObserver` to the `PreactBaseElement`
2. I don't know if we really win anything from `attributeFilter` for a single element. They are mainly useful for `subtree` observers. But if anyone feels strongly about it, we can extract all attributes from  the `AmpFitText['props']`.
3. The mutation observer should simply call `mutatedAttributesCallback()` - that's what we expected in the first place.

Reply:
We'll get double `mutatedAttributesCallback` calls when the mutation comes from amp-bind, though. We'll need a way to prevent that.

Reply:
Adding this as a followup item in #28281 since it sounds like more discussion is needed here.